model_api/cpp/utils/include/utils/async_infer_queue.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+// Copyright (C) 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include <condition_variable>
+#include <functional>
+#include <queue>
+#include <vector>
+
+#include <openvino/openvino.hpp>
+
+
+class AsyncInferQueue {
+public:
+    AsyncInferQueue() = default;
+    AsyncInferQueue(ov::CompiledModel& model, size_t jobs);
+    ~AsyncInferQueue();
+    bool is_ready();
+    size_t get_idle_request_id();
+    void wait_all();
+    void set_default_callbacks();
+    void set_custom_callbacks(std::function<void(ov::InferRequest, std::shared_ptr<ov::AnyMap> callback_args)> f_callback);
+    size_t size() const;
+    void start_async(const ov::Tensor& input, std::shared_ptr<ov::AnyMap> userdata = nullptr);
+    void start_async(const std::map<std::string, ov::Tensor>& input, std::shared_ptr<ov::AnyMap> userdata = nullptr);
+    ov::InferRequest operator[](size_t i);
+
+    // AsyncInferQueue is the owner of all requests. When AsyncInferQueue is destroyed,
+    // all of requests are destroyed as well.
```

Comment:
You can't really control that. `ov::InferRequest` is just a shared_ptr.  One can get an `ov::InferRequest` with `operator[]()` and copy it. After the `m_requests` is destroyed, the internals of `ov::InferRequest` will continue living. I think you should remove the comment or leave an explanation similar to that.

Reply:
I see, the comments came from the original implementation. I'll update it